src/core/swap.c

```diff
@@ -374,11 +374,10 @@ static int swap_setup_unit(
                 int priority,
                 bool set_flags) {
 
+        _cleanup_(unit_freep) Unit *new = NULL;
```

Comment:
hmm, this variable always goes out of scope and is destroyed. There's some TAKE_PTR(new) missing below in the succcess path?

Reply:
ahh, this part already got fixed in abef0ec308dc2adf90798733902bff23a7e3a916. Ignore this comment hence.